database/src/desktop/connection/web_socket_client_impl.cc

```diff
@@ -45,7 +46,8 @@ WebSocketClientImpl::WebSocketClientImpl(
       user_agent_(user_agent),
       logger_(logger),
       scheduler_(scheduler),
-      safe_this_(this) {
+      safe_this_(this),
+      app_check_token_(app_check_token) {
```

Comment:
So normally if updates the connection by sending a different message on the existing connection to update the token, which is handled in persistent_connection::RefreshAppCheckToken.  However, you are right that if the connection gets closed and then re-opened, it could be a problem.  I'm not 100% sure that is possible, it seems like it usually just destructs and reconstructs if the connection is interrupted, but I'm also not sure that always happens, so better safe than sorry.

Reply:
Oh I missed RefreshAppCheckToken updating the client_impl app check token. That makes sense.